java/src/jni/h5sImp.c

```diff
@@ -1340,6 +1340,389 @@ Java_hdf_hdf5lib_H5_H5Sget_1regular_1hyperslab(JNIEnv *env, jclass clss, jlong s
+            CHECK_JNI_EXCEPTION(ENVONLY, JNI_TRUE);
+            H5_BAD_ARGUMENT_ERROR(ENVONLY, "H5Sselect_hyperslab: stride array length < 0");
+        }
+
+        if (stride_rank != start_rank)
+            H5_BAD_ARGUMENT_ERROR(ENVONLY, "H5Sselect_hyperslab: stride and start have different rank!");
+
+        PIN_LONG_ARRAY(ENVONLY, stride, strideP, &isCopy, "H5Sselect_hyperslab: stride not pinned");
+
+        if (NULL == (strd = lp = (hsize_t *)HDmalloc((size_t)stride_rank * sizeof(hsize_t))))
+            H5_OUT_OF_MEMORY_ERROR(ENVONLY, "H5Sselect_hyperslab: failed to allocate stride buffer");
+
+        jlp = (jlong *)strideP;
+        for (i = 0; i < stride_rank; i++) {
+            *lp = (hsize_t)*jlp;
+            lp++;
+            jlp++;
+        } /* end if */
+    }
+
+    if (NULL == block) {
+        blockP = NULL;
+        blk    = (hsize_t *)blockP;
+    }
+    else {
+        if ((block_rank = ENVPTR->GetArrayLength(ENVONLY, block)) < 0) {
+            CHECK_JNI_EXCEPTION(ENVONLY, JNI_TRUE);
+            H5_BAD_ARGUMENT_ERROR(ENVONLY, "H5Sselect_hyperslab: block array length < 0");
+        }
+
+        if (block_rank != start_rank)
+            H5_BAD_ARGUMENT_ERROR(ENVONLY, "H5Sselect_hyperslab: block and start have different rank!");
+
+        PIN_LONG_ARRAY(ENVONLY, block, blockP, &isCopy, "H5Sselect_hyperslab: block not pinned");
+
+        if (NULL == (blk = lp = (hsize_t *)HDmalloc((size_t)block_rank * sizeof(hsize_t))))
+            H5_OUT_OF_MEMORY_ERROR(ENVONLY, "H5Sselect_hyperslab: failed to allocate block buffer");
+
+        jlp = (jlong *)blockP;
+        for (i = 0; i < block_rank; i++) {
+            *lp = (hsize_t)*jlp;
+            lp++;
+            jlp++;
+        } /* end for */
+    }
+
+    if ((sid = H5Scombine_hyperslab(space_id, (H5S_seloper_t)op, (const hsize_t *)strt, (const hsize_t *)strd,
+                                    (const hsize_t *)cnt, (const hsize_t *)blk)) < 0)
+        H5_LIBRARY_ERROR(ENVONLY);
+
+done:
+    if (blk)
+        HDfree(blk);
+    if (blockP)
+        UNPIN_LONG_ARRAY(ENVONLY, block, blockP, JNI_ABORT);
+    if (strd)
+        HDfree(strd);
+    if (strideP)
+        UNPIN_LONG_ARRAY(ENVONLY, stride, strideP, JNI_ABORT);
+    if (cnt)
+        HDfree(cnt);
+    if (countP)
+        UNPIN_LONG_ARRAY(ENVONLY, count, countP, JNI_ABORT);
+    if (strt)
+        HDfree(strt);
+    if (startP)
+        UNPIN_LONG_ARRAY(ENVONLY, start, startP, JNI_ABORT);
+
+    return (jlong)sid;
+} /* end Java_hdf_hdf5lib_H5_H5Sselect_1hyperslab */
+
+/*
+ * Class:     hdf_hdf5lib_H5
+ * Method:    H5Smodify_select
+ * Signature: (JIJ)V
+ */
+JNIEXPORT void JNICALL
+Java_hdf_hdf5lib_H5_H5Smodify_1select(JNIEnv *env, jclass clss, jlong space1_id, jint op, jlong space2_id)
+{
+    hssize_t retVal = -1;
+
+    UNUSED(clss);
+
+    if ((retVal = H5Smodify_select((hid_t)space1_id, (H5S_seloper_t)op, (hid_t)space2_id)) < 0)
```

Comment:
The error should be handled by the         H5_LIBRARY_ERROR(ENVONLY);

Reply:
I don't think that's normally the way that we handle errors though. I briefly looked at other routines that return herr_t and they usually have jint or similar as a return value. No reason to hide the error by making it only show up in the error stack and not as an actual return value.

Reply:
Yes it is since 1.10. The old way was to return the int, which was useless because the exception would happen before you could check the return value.